### PR TITLE
Spag Environments

### PR DESCRIPTION
**Pull request by Squab**
* Adds key-value based environments that are written to a file at .spag/environments
* Default environment is used when none is specified
* Set and unset key-value pairs in the environment
* Show command so that environment can be examined
* Updating environments with set command updates the env file

### CODE REVIEW

File: spag_files.py [hunk 1]
```diff
@@ -107,3 +106,111 @@ def get_file_list(self):
...
                 for path in paths]
+
+class SpagEnvironment(object):
+
+    SPAG_ENV_DIR = './.spag/environments/'
+    DEFAULT_ENV_NAME = 'default.yml'
+
+    @classmethod
+    def activate(cls, envname):
+        filename = cls.SPAG_ENV_DIR + envname + '.yml'
```
Comment (pglass) [inline spag_files.py]: Can you `os.path.join(cls.SPAG_ENV_DIR, envname + '.yml')` for these?

Comment (Squab) [inline spag_files.py]: Sure

**Comment by pglass** [comment]
- You can use `spag_remembers.ensure_dir_exists` which is the same as `mkdir -p` to make the environment dir. Or we can just add `spag init` later.
- For the envvars, I think we should use `-E thing=val` instead of `-E thing:val`. We'll do the same thing for the `--with thing_id=poo` and can reuse that code in both places.
- I think whitespace is not allowed in http headers? e.g. if I do `requests.post(http://localhost:5000/headers, headers={'poo': '    abcde  '})`, then I get back `{'poo': 'abcde'}`. I think we can `str.strip()` all the whitespace from the header key and value. (try `-H "poo:   wumbo"` and the environment will have a string with whitespace in the header).

File: tests/test_spag.py [hunk 2]
```diff
@@ -236,14 +199,13 @@ def test_spag_load_file(self):
 
     def test_spag_request_get(self):
         for name in ('auth.yml', 'auth'):
-            out, err, ret = run_spag('request', name, '--dir', RESOURCES_DIR)
-            self.assertEqual(err, '')
-            self.assertEqual(json.loads(out), {"token": "abcde"})
+            out, err, ret = run_spag('request', 'auth.yml')
```
Comment (pglass) [inline tests/test_spag.py]: This needs `name` here instead of `auth.yml`.

File: tests/test_spag.py [hunk 3]
```diff
@@ -263,33 +224,31 @@ def test_spag_request_patch(self):
...
             self.assertEqual(ret, 0)
 
-            out, err, ret = run_spag('request', name, '--dir', RESOURCES_DIR)
-            self.assertEqual(ret, 0)
-            self.assertEqual(out, '\n')
-            self.assertEqual(err, '')
+        out, err, ret = run_spag('request', 'delete_thing.yml')
+        self.assertEqual(ret, 0)
+        self.assertEqual(out, '\n')
+        self.assertEqual(err, '')
```
Comment (pglass) [inline tests/test_spag.py]: This last block either needs to be inside the loop (with `name` instead of `delete_thing.yml`) or only make one post request.

**Comment by pglass** [comment]
Okay. So I know I said we should use `key=value`... But I feel like the headers should still use a colon. I guess it doesn't matter though.

**Comment by Squab** [comment]
@pglass Done :)

**Pull request merged**